src/test/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabPipelineStatusNotifierTest.java

```diff
@@ -40,12 +40,46 @@ public void should_set_branch_status_name() {
                 + "branch"));
     }
 
+    @Test
+    public void should_set_branch_status_name_withBuildStatusNameCustomPart() {
+        GitLabSCMSourceContext sourceContext = new GitLabSCMSourceContext(null, null);
+        sourceContext.withBuildStatusNameCustomPart("CUSTOM");
+        sourceContext.withBuildStatusNameOverwrite(false);
+
+        BranchSCMHead head = new BranchSCMHead("head");
+        SCMRevision revision = new BranchSCMRevision(head, "hash");
+
+        String statusName = GitLabPipelineStatusNotifier.getStatusName(sourceContext, null, revision,
+                new hudson.EnvVars());
+
+        assertThat(statusName, is(GitLabPipelineStatusNotifier.GITLAB_PIPELINE_STATUS_PREFIX
```

Comment:
Maybe instead of referencing constants here, write the actual literal expected value. Changing the value is an API break essentially?

Started external GitLab pipeline can't finish if the plugin is updated in between - jenkins pipelines can "in theory" run across Jenkins restarts...

Reply:
We had a similar issue when changing the project config *Discover merge requests from forks*:

* old value: `Discover merge requests from forks` 
* new value: `The current merge request revision`

That has the side effect to report the job status to be reported as `jenkinsci/mr-head` on GitLab instead of  `jenkinsci/mr-merge`.

You also end up with tons of pipeline `jenkinsci/mr-merge` the are "running" forever:

![External pipeline in Jenkins](https://github.com/jenkinsci/gitlab-branch-source-plugin/assets/1222165/f3df3389-c545-49d4-bc21-1cb939af1bce)

For me you just need to accept that and be prepared to run some `curl` command to make the pipeline green using the REST API: https://docs.gitlab.com/ee/api/commits.html#set-the-pipeline-status-of-a-commit

```
curl --location '****gitlab-server****/api/v4/projects/****project-id****/statuses/****commit-sha****' \
--header 'Content-Type: application/json' \
--header 'PRIVATE-TOKEN: ****************' \
--data '{
    "state": "success",
    "name": "jenkinsci/mr-merge"
}'

Reply:
This PR is an attempt to resolve such problems by removing the type from the pipeline name... because GitLab is not GitHub and - depending on how you use the branch source - the type may not make sense.

Reply:
I am well aware about what this change is about.

We are interested to have a single name instead of:
`jenkinsci/branch` and `jenkins/mr-head` 

In order to avoid 2 external pipeline report in Jenkins…